### PR TITLE
Add more documentation features

### PR DESCRIPTION
**Pull request by FredTheDino**
Document the entirety of Snake-Ribs and add a table of content.

### CODE REVIEW

File: ribs.docs [hunk 1]
```diff
@@ -24,3 +26,112 @@ if key_down(pg.K_LEFT):
...
+# `overlap_data(a, b)`
+Returns the exis that points from a to b, and the depth of the collision.
+If the depth is negative it means they are far away from overlapping.
+
+Note that this function is not super important, you probably want
+to use `solve_rect_overlap(a, b)`
+
+<div class="arg-block">
+<div class="arg-name">`a`</div>- A `pg.Rect(minx, miny, width, height)`.<br>
+<div class="arg-name">`b`</div> - A `pg.Rect(minx, miny, width, height)`. (Same as `a`)<br>
```
Comment (sornas) [inline ribs.docs]:
```suggestion
<div class="arg-name">`a`</div> - A `pg.Rect(minx, miny, width, height)`.<br>
<div class="arg-name">`b`</div> - A `pg.Rect(minx, miny, width, height)`. (Same as `a`)<br>
```

File: ribs.docs [hunk 2]
```diff
@@ -24,3 +26,112 @@ if key_down(pg.K_LEFT):
...
+<div class="arg-name">`a`     </div> - A `pg.Rect(minx, miny, width, height)`.<br>
+<div class="arg-name">`b`     </div> - A `pg.Rect(minx, miny, width, height)`. (Same as `a`)<br>
+<div class="arg-name">`vel_a` </div> - The velocity `a` has, given as a tuple.<br>
+<div class="arg-name">`vel_b` </div> - The velocity `b` has, given as a tuple.<br>
+<div class="arg-name">`mass_a`</div> - The mass of `a`, 0 mass means it's static and imovable.<br>
+<div class="arg-name">`mass_b`</div> - The mass of `b`, 0 mass means it's static and imovable.<br>
+<div class="arg-name">`bounce`</div> - How bouncy the collision should be. Values higher than 1 will introduce
+           more energy to the system and might cause glitches. Values lower than 0
+           does not make sense.<br>
+</div>
```
Comment (sornas) [inline ribs.docs]: I feel like the documentation generator could handle this better than writing it by hand. Something like:

```
- a: A `pg.Rect(minx, miny, width, height)
- b: ...
- vel_a: The velocity `a` has, given as a tuple.
```
And so on. The format is just an example, you could also go more explicit with something like `[vel_a] The velocity ...`.

File: ribs.docs [hunk 3]
```diff
@@ -24,3 +26,112 @@ if key_down(pg.K_LEFT):
...
+<div class="arg-block">
+<div class="arg-name">`key`</div>- Is a string with the character or a keycode (pg.K_****) of the button to check.<br>
+</div>
+
+## ex
+!--
+if key_released("a"):
+    print("hold longer maybe?")
+--!
+this code would print "hold longer maybe?" to the console when the "a" button was released.
```
Comment (sornas) [inline ribs.docs]:
```suggestion
This code would print "hold longer maybe?" to the console when the "a" button was released.
```

File: ribs.docs [hunk 4]
```diff
@@ -24,3 +26,112 @@ if key_down(pg.K_LEFT):
...
+
+## ex
+!--
+if key_down(pg.K_SPACE):
+    print("JUMP")
+--!
+This code would print "JUMP" to the console when the spacebar is pressed, and only that frame.
+
+# `draw_transformed(img, position, scale=(1., 1.), degrees=0)`
+Draws a sprite in a sprite centered around `position` scaled by `scale` and
```
Comment (sornas) [inline ribs.docs]:
```suggestion
Draws a sprite centered around `position` scaled by `scale` and
```

File: ribs.docs [hunk 5]
```diff
@@ -24,3 +26,112 @@ if key_down(pg.K_LEFT):
...
+rotated clockwise `degrees`. This function is just a convinience function to
```
Comment (sornas) [inline ribs.docs]:
```suggestion
rotated clockwise `degrees`. This function is just a convenience function to
```

File: ribs.docs [hunk 6]
```diff
@@ -24,3 +26,112 @@ if key_down(pg.K_LEFT):
...
+<div class="arg-name">`degrees`</div> - If you want to rotate the sprite. (You don't need to supply this argument.)<br>
+</div>
+
+## ex
+!--
+assets["teapot"] = pg.image.load("teapot.png")
+draw_transformed(assets["teapot"], (100, 100))
+--!
+This code would draw the "teapot.png" image (given that you have one) 100
+pixels to the right of the left edge, and 100 pixels bellow the top edge of the
```
Comment (sornas) [inline ribs.docs]:
```suggestion
pixels to the right of the left edge, and 100 pixels below the top edge of the
```

File: ribs.docs [hunk 7]
```diff
@@ -24,3 +26,112 @@ if key_down(pg.K_LEFT):
...
+window.
+
+## ex
+!--
+assets["teapot"] = pg.image.load("teapot.png")
+draw_transformed(assets["teapot"], (100, 100), (1, 0.5), time)
+--!
+This code would draw the "teapot.png" image (given that you have one),
+and given that `time` was the "time since the program started", this
+would draw a sprite at (100, 100) shrunk to half long the y axis, rotating
```
Comment (sornas) [inline ribs.docs]:
```suggestion
would draw a sprite at (100, 100) shrunk to half size along the y axis, rotating
```

File: ribs.docs [hunk 8]
```diff
@@ -24,3 +26,112 @@ if key_down(pg.K_LEFT):
...
+around it's center.
+
+# `overlap_data(a, b)`
+Returns the exis that points from a to b, and the depth of the collision.
+If the depth is negative it means they are far away from overlapping.
```
Comment (sornas) [inline ribs.docs]: exis -> axis? I think.

File: ribs.docs [hunk 9]
```diff
@@ -24,3 +26,112 @@ if key_down(pg.K_LEFT):
...
+## ex
+!--
+rect_a = pg.Rect(200, 200, 100, 100)
+vel_a = (0, 1000)
+rect_b = pg.Rect(250, 250, 50, 100)
+vel_b = (0, 0)
+rect_a, rect_b, vel_a, vel_b, hit = solve_rect_overlap(rect_a, rect_b, vel_a, vel_b, bounce=1.0)
+--!
+This code will create a collision between `rect_a` and `rect_b`. The collision will send
+rect_b flying since all the energy from `rect _a` is transfered to `rect_b`.
```
Comment (sornas) [inline ribs.docs]:
```suggestion
rect_b flying since all the energy from `rect _a` is transferred to `rect_b`.
```

File: ribs.docs (hunk 2 above)
Comment (sornas) [reply ribs.docs]: #9

File: ribs.docs (hunk 1 above)
Comment (FredTheDino) [reply ribs.docs]: This is solved by PR 9

File: ribs.docs (hunk 2 above)
Comment (FredTheDino) [reply ribs.docs]: I agree, I just couldn't be bothered with doing it... xD

**Comment by FredTheDino** [comment]
Rebase and fixup these commits after #9  is merged in

**Review by sornas — APPROVED** [review]

**Pull request merged**